src/org/jgroups/protocols/raft/FileBasedLog.java

```diff
@@ -166,11 +172,42 @@ public int lastAppended() {
    }
 
    public void setSnapshot(ByteArray sn) {
-      snapshot=sn;
+      Path snapshotPath = snapshotPath();
+      try {
+         if (Files.exists(snapshotPath)) {
+            // write to temporary file first
+            Path tmp = Files.createTempFile(logDir.toPath(), null, null);
+            writeSnapshot(sn, tmp);
+            // do we need atomic move?
+            Files.move(tmp, snapshotPath, StandardCopyOption.REPLACE_EXISTING, StandardCopyOption.ATOMIC_MOVE);
+         } else {
+            writeSnapshot(sn, snapshotPath);
```

Comment:
If this is supposed to be durable - or survive to power loss, it should fsync on the parent folder as well

Reply:
Yes, it's supposed to be durable

Reply:
I don't understand why we need fsync in the parent directory. Isn't fsync in the file enough?